tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/server/ExpireQueueSuffixTest.java

```diff
@@ -0,0 +1,101 @@
+import javax.jms.ConnectionFactory;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+
+import org.apache.activemq.artemis.api.core.QueueConfiguration;
+import org.apache.activemq.artemis.api.core.RoutingType;
+import org.apache.activemq.artemis.api.core.SimpleString;
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.apache.activemq.artemis.core.server.Queue;
+import org.apache.activemq.artemis.core.settings.impl.AddressSettings;
+import org.apache.activemq.artemis.tests.util.ActiveMQTestBase;
+import org.apache.activemq.artemis.tests.util.CFUtil;
+import org.apache.activemq.artemis.tests.util.Wait;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class ExpireQueueSuffixTest extends ActiveMQTestBase {
+
+   public final SimpleString queueA = SimpleString.of("queueA");
+   public final SimpleString queueB = SimpleString.of("queueB");
+   public final SimpleString expiryAddress = SimpleString.of("myExpiry");
+
+   public final SimpleString expirySuffix = SimpleString.of(".expSuffix");
+   public final long EXPIRY_DELAY = 10L;
+
+   private ActiveMQServer server;
+
+   @Override
+   @BeforeEach
+   public void setUp() throws Exception {
+      super.setUp();
+      server = createServer(true, true);
+      server.getConfiguration().setAddressQueueScanPeriod(50L).setMessageExpiryScanPeriod(50L);
+
+      server.getAddressSettingsRepository().addMatch("#", new AddressSettings().setAutoCreateExpiryResources(true).setExpiryAddress(expiryAddress).setExpiryDelay(EXPIRY_DELAY).setExpiryQueueSuffix(expirySuffix));
+
+      server.start();
+
+      server.createQueue(QueueConfiguration.of(queueA).setRoutingType(RoutingType.ANYCAST));
+      server.createQueue(QueueConfiguration.of(queueB).setRoutingType(RoutingType.ANYCAST));
+   }
+
+   @Test
+   public void testAutoCreationOfExpiryResources() throws Exception {
+      ConnectionFactory factory = CFUtil.createConnectionFactory("CORE", "tcp://localhost:61616");
+
+      long sendA = 7;
+      long sendB = 11;
+
+      try (Connection connection = factory.createConnection()) {
+         Session session = connection.createSession(true, Session.SESSION_TRANSACTED);
+         MessageProducer producer = session.createProducer(session.createQueue(queueA.toString()));
+         producer.setTimeToLive(100);
+
+         for (int i = 0; i < sendA; i++) {
+            producer.send(session.createTextMessage("queueA"));
+         }
+         session.commit();
+
+         producer = session.createProducer(session.createQueue(queueB.toString()));
+         producer.setTimeToLive(100);
+         for (int i = 0; i < sendB; i++) {
+            producer.send(session.createTextMessage("queueB"));
+         }
+         session.commit();
+      }
+
+      Wait.waitFor(() -> server.locateQueue(expiryAddress.toString(), "EXP." + queueA + expirySuffix) != null, 5000);
```

Comment:
Hehe, yes, on my earlier comment about using variables for clarity...something a lot like this. Could have saved myself some typing by reading ahead and referencing this hehe.

(A variable could perhaps be added for this whole name to make it easier to follow)